content/en/blog/2023/otel-dotnet-performance-impacts.md

```diff
@@ -0,0 +1,287 @@
+wrote, and also this post.
+
+## Baseline test
+
+What does a good test look like? All code is different, so it’s hard to put a
+finger on some representative code. What I will say is that you need some
+“actual code” to know whether the compiler is optimizing some things. This also
+serves to give you some confidence that this is how the code would react in a
+real application.
+
+Let’s start with something really basic. This code will check if something
+exists in a dictionary, and return the value.
+
+```csharp
+public class TestBase
+{
+    internal readonly ActivitySource _source = new("PerfTest");
+
+    private readonly Dictionary<string, string> _dictionary = new();
+
+    public TestBase()
+    {
+        _dictionary = new Dictionary<string, string> {
+            { "blah", "blah" },
+            { "blah1", "blah" }
+        };
+    }
+
+    private string InternalTest()
+    {
+        if (_dictionary.ContainsKey("blah"))
+            return _dictionary["blah"];
+
+        return "";
+    }
+
+}
+```
+
+This should give us a nice baseline for something that has some actual
+computation. We can then build on it to see how adding telemetry affects
+response time.
+
+## The tests
+
+What I’m trying to achieve is to answer the question, “What if you start with
+nothing?” Then, to build up what parts of instrumentation affect your code—and
+establish the impact.
+
+We’re looking at these specific questions:
+
+1. **Does adding an `Activity` that isn’t listened to cause additional latency
+   in calling the method?** This is to help answer the questions library authors
+   have around adding `Activity` as a means to provide debug information to
+   consumers of their library.
+1. **Does listening to an `Activity` using OpenTelemetry cause additional
+   latency in calling the method?** This is to help developers understand the
+   impact of turning on different elements of telemetry for libraries they’re
+   using.
+1. **How do “Processors” in OpenTelemetry affect the performance of the
+   application?** This is to help clarify the impact application developers can
+   have on application code by augmenting telemetry.
+1. **Does adding more “Tags” (Attributes) for an `Activity` affect an
+   application’s performance?** This helps engineers understand the impact of
+   adding more context to their tracing.
+
+The overall hypothesis I’ve used for this analysis is that the effects are
+negligible and should not be considered as a reason not to use them.
+
+For the scope of this test, I’ll be running through WSL2, using Ubuntu and .NET
+7.0. It’s an 11th Gen i7 (Surface Laptop Studio).
+
+```
+BenchmarkDotNet=v0.13.3, OS=ubuntu 22.10
+11th Gen Intel Core i7-11370H 3.30GHz, 1 CPU, 8 logical and 4 physical cores
+.NET SDK=7.0.100
+  [Host]     : .NET 7.0.0 (7.0.22.51805), X64 RyuJIT AVX2
+  DefaultJob : .NET 7.0.0 (7.0.22.51805), X64 RyuJIT AVX2
+```
+
+## The Results
```

Comment:
please include the code used for this benchmarks.

Reply:
The code is inline in the document.

Reply:
it's in each section and describes what it's doing. There isn't a repo for it, it's not part of this repo.

Reply:
There is no code shared anywhere in this blog. It describes something, but without code, its not possible to tell what exactly is the benchmarks doing and how.